### DISCUSSION TITLE
Does
hyperview support fingerprint authentication on the mobile device?

### DISCUSSION THREAD

**Discussion by hrstoyanov** [Instawork/hyperview#858] [Q&A] [answered]
Or do I need an extension for that? 
Thanks.

**Comment by adamstep** [comment]
Hi @hrstoyanov , this is not supported in the core library. Our design philosophy is to keep the core lean, and extra functionality can be added through extensions (custom components and behaviors). A quick look at available libraries shows the existence of https://www.npmjs.com/package/react-native-fingerprint-scanner , which could be wrapped in a custom component that can trigger events based on success/failure to authenticate a fingerprint.

**Reply by hrstoyanov** [reply]
@adamstep ok, I see, but just like  with #853, this is arguably not an optional feature these days - passwordless authentication is a big thing for 2024 and for hyperview to stay competitive. Developers who came here for the hypermedia approach will find themselves juggling react native code and extensions...

**Reply by adamstep** [reply, marked as answer]
I agree it's not ideal for developers to need to write React Native code when they just want to build a hypermedia-driven application. However, we can't support every desired feature in the core Hyperview library for two main reasons:

- the burden of maintaining these features is higher than we can support. We want to spend our time making sure the core library is bug-free, performant, and flexible enough to support extensions.
- if we add every feature requested by the community to the core, the framework will become bloated, leading to larger app binary sizes and slower startup times. These two metrics are very important to mobile app UX, and we don't want developers to pay a penalty for features they don't need.

What we're trying to encourage is a selection of community-authored components that can be easy to integrate. For example, if someone creates and shares a `hyperview-fingerprint-auth` extension, then including it in your app would be relatively simple (add the package dependency, and update one line of config to register the extension in your app). We have plans to contribute many of Instawork's extensions to the community to get the ball rolling, including our QR-code reading component. Stay tuned for more over the rest of the year.
